### ISSUE TITLE
Add allow_generics_not_specified

### ISSUE THREAD

**Issue by Thiltal** [dart-lang/linter#205]
Add possibility to not require generics

In my project, I want to require type annotations, but f.e. Map is sufficient for me.

**Comment by Hixie** [comment]
Probably better to just separate the existing lint into two, one for requiring that generics be explicit, and one for requiring that everything else be explicit.

**Comment by bwilkerson** [comment]
Agreed. It would be good to avoid having configuration options on lints if possible.

**Comment by pq** [comment]
> Add possibility to not require generics
> 
> In my project, I want to require type annotations, but f.e. Map is sufficient for me.

I'm assuming you mean type declarations?  IOW:

```Map m = new HashMap();```

In case you were referring to instance checks, it's worth pointing out that `is` checks are now fixed so that

```
if(x is Map) {
  ...
}
```

is accepted.  (See https://github.com/dart-lang/linter/issues/227.)

**Comment by zoechi** [comment]
Not sure if @Hixie means this but I added a lot of `<String,dynamic>` to `{'a': 3}` or `Map foo(Map x) => ...` because I can't apply 
```dart
@optionalTypeArgs
``` 
to `Map` or other code I don't control.

**Comment by Hixie** [comment]
I just meant that the lint requiring that `foo(i) => i + 1;` be `int foo(int i) => i + 1;` should be separate from the lint requiring that `Foo bar = new Foo();` be `Foo<int> bar = new Foo<int>();`. I'd certainly also support even finer-grain lints to control whether literals should have types, etc. Just so long as there's a way to have all types required, and a way to have all the other desires people have be met. :-)

**Comment by pq** [comment]
Closing this one as stale.  Feel free to re-open if you want to continue the discussion.  Thanks!
